src/plugins/vis_type_timeseries/public/application/components/lib/check_if_series_have_same_formatters.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { last } from 'lodash';
+import { DATA_FORMATTERS } from '../../../../common/enums';
+import type { Series } from '../../../../common/types';
+import type { FieldFormatMap } from '../../../../../data/common';
+
+export const checkIfSeriesHaveSameFormatters = (
+  seriesModel: Series[],
+  fieldFormatMap?: FieldFormatMap
+) => {
+  const allSeriesHaveDefaultFormatting = seriesModel.every(
+    (seriesGroup) => seriesGroup.formatter === DATA_FORMATTERS.DEFAULT
+  );
+
+  return allSeriesHaveDefaultFormatting && fieldFormatMap
+    ? seriesModel
```

Comment:
nit: for me this code can be refactored and instead of your 3 iterations of `seriesModel` array you can combine it into one `seriesModel.reduce`.